### PR TITLE
feat: implement googlepay payment method

### PR DESCRIPTION
**Pull request by pfrincu-plenty**
## Why:

Closes: #ID

## Checklist before requesting a review

- [ ] My code follows the code style of this project.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I've updated `docs/changelog/changelog_en.md` and `docs/changelog/changelog_de.md`. If I'm a non-German speaker, I've still updated the file with the English version as a placeholder.

### CODE REVIEW

File: apps/web/components/PayPal/GooglePayButton.vue [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+  </div>
+  <div id="google-pay-button"></div>
+</template>
+
+<script lang="ts" setup>
+import { PayPalAddToCartCallback } from '~/components/PayPal/types';
+import { cartGetters, orderGetters } from '@plentymarkets/shop-api';
+
+let isGooglePayLoaded = true;
+let countryCodeString = 'DE';
```
Comment (FabianGerke) [inline apps/web/components/PayPal/GooglePayButton.vue]: isGooglePayLoaded should be `false` by default and a ref().
Country code is hardcoded that should be fixed as well.

File: apps/web/components/PayPal/GooglePayButton.vue [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+  };
+  const paymentDataRequest = Object.assign({}, baseRequest);
+  paymentDataRequest.allowedPaymentMethods = allowedPaymentMethods;
+  paymentDataRequest.transactionInfo = getGoogleTransactionInfo();
+  paymentDataRequest.merchantInfo = merchantInfo;
+  paymentDataRequest.callbackIntents = ['PAYMENT_AUTHORIZATION'];
+  return paymentDataRequest;
+}
+
+function onPaymentAuthorized(paymentData: any): Promise<google.payments.api.PaymentAuthorizationResult> {
```
Comment (FabianGerke) [inline apps/web/components/PayPal/GooglePayButton.vue]: Shouldn't be type any

File: apps/web/components/PayPal/GooglePayButton.vue [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+  emits('button-clicked', async (successfully) => {
+    if (successfully) {
+      const paymentDataRequest = await getGooglePaymentDataRequest();
+      const paymentsClient = getGooglePaymentsClient();
+      paymentsClient.loadPaymentData(paymentDataRequest);
+    }
+  });
+}
+
+async function processPayment(paymentData: any) {
```
Comment (FabianGerke) [inline apps/web/components/PayPal/GooglePayButton.vue]: Shouldn't be type any

File: apps/web/components/PayPal/GooglePayButton.vue [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+      paymentMethodData: paymentData.paymentMethodData,
+    });
+
+    if (status === 'PAYER_ACTION_REQUIRED') {
+      // eslint-disable-next-line promise/catch-or-return
+      (paypal as any)
+        .Googlepay()
+        .initiatePayerAction({ orderId: order.order.id })
+        // eslint-disable-next-line promise/always-return
+        .then(async (data: any) => {
```
Comment (FabianGerke) [inline apps/web/components/PayPal/GooglePayButton.vue]: Shouldn't be type any

File: apps/web/components/PayPal/GooglePayButton.vue [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+        mode: 'paypal',
+        plentyOrderId: Number.parseInt(orderGetters.getId(order)),
+        paypalTransactionId: transaction.id,
+      });
+    }
+    clearCartItems();
+    navigateTo(localePath(paths.confirmation + '/' + order.order.id + '/' + order.order.accessKey));
+
+    return { transactionState: 'SUCCESS' };
+  } catch (error: any) {
```
Comment (FabianGerke) [inline apps/web/components/PayPal/GooglePayButton.vue]: Shouldn't be type any

**Review by FabianGerke — CHANGES_REQUESTED** [review]

File: apps/web/components/PayPal/GooglePayButton.vue (hunk 1 above)
Comment (pfrincu-plenty) [reply apps/web/components/PayPal/GooglePayButton.vue]: We couldn't find any device which doesn't accept Googlepay, that's why we set it always true

**Pull request closed without merging**